nodes.py

```diff
@@ -221,7 +217,50 @@ def check_semantics(self, variables_dict):
 
     def __str__(self, indent_level=0):
         return super().__str__(indent_level)
-
+    
+    def write_code(self, output_lines):
+        var_type, var_value, var_mem_id = ProgramMemory.variables_dict[self.left.name]
+        right_type, right_mem_id, right_value = self.right.write_code(output_lines)    
+        if var_type is Types.Int:
+            if right_value != "":
+                output_lines.append(
+                    f"  store i32 {right_value}, i32* %{var_mem_id}, align 4"
+                )
+            else:
+                output_lines.append(
+                    f"  store i32 %{right_mem_id}, i32* %{var_mem_id}, align 4"
+                )
+        if var_type is Types.Float:
+            if right_type is Types.Int:
+                if right_value != "":
+                    output_lines.append(
+                        f"  %{ProgramMemory.mem_counter} = sitofp i32 {right_value} to float"
```

Comment:
I LLVM type name for it is called  "double" not "float". I checked it with clang to confirm it. Not only this method but whole BinOp class requires changing from "float" to "double".

Reply:
As far as I know there is float type which uses 4 bytes, but I can change it to double with 8 bytes.

Reply:
Changed float to double.